src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,165 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(last, value, null);
+        if (size == 0) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if (size == index) {
+            add(value);
+        } else {
+            Node<T> newNode = new Node<>(null, value, null);
+            Node<T> currentNode = getCurrentNode(index);
+            if (index == 0) {
+                newNode.next = currentNode;
+                currentNode.prev = newNode;
+                first = newNode;
+            } else {
+                currentNode.prev.next = newNode;
+                newNode.prev = currentNode.prev;
+                newNode.next = currentNode;
+                currentNode.prev = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value: list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = getCurrentNode(index);
+        return currentNode.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = getCurrentNode(index);
+        T oldValue = currentNode.value;
+        currentNode.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = getCurrentNode(index);
+        unlink(index);
+        return currentNode.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        int currentIndex = -1;
+        Node<T> currentNode = first;
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(currentNode.value, object)) {
+                currentIndex = i;
+                break;
+            }
+            currentNode = currentNode.next;
+        }
+        if (currentIndex == -1) {
+            return false;
+        }
+        unlink(currentIndex);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void unlink(int index) {
+        Node<T> currentNode = getCurrentNode(index);
+        if (index == 0 && size == 1) {
+            first = null;
+            last = null;
+        } else {
+            if (index == 0) {
```

Comment:
Try to avoid such constructions ('if' inside another 'if'). It makes code more difficult to read
if (...) {
    if (...) {
    }  
}